test/browser.test.js

```diff
@@ -18,6 +19,10 @@ describe('browser', function() {
     exec('node --eval "require(\'./lib/browser\')"', done);
   });
 
+  it('require() works in an environment where process.versions is an empty object (gh-5842)', function(done) {
+    exec('node --eval "const process = require(\\"node:process\\"); process.versions = {}; require(\'./lib/browser\')"', done);
```

Comment:
Make this `require('process')`. Older node versions don't support `require('node:process')`

Reply:
Sure, will do. This test was just a placeholder as it didn't properly verify the issue.
I found that `Object.defineProperty(process, 'versions', { value: {} })` works for replacing the versions object. I pushed an updated version (still without the `null` handling described in the other comment).

Reply:
There is another issue connected to it.
If you do `require('./lib.browser.js')` with `process.versions` replaced with `{}` there is another piece of code that fails. It is located in one of the dependent libraries (`bluebird` required by `mquery`).
It seems to be removed by tree shaking done by webpack as it is not present in the final browser bundle.
Thus, I made the test load the bundle file (`./dist/browser.umd.js`) not `./lib/browser`. It may lead to additional quirks when testing locally as the file must be built before running tests.